drbd-kernel-module-injector fails when using drbd9-centos7:v9.0.23 on CentOS 7.8

Compile fails when using drbd9-centos7:v9.0.23 on CentOS 7.8

```
# kubectl logs -f piraeus-op-ns-node-mf7zf -c drbd-kernel-module-injector
Need a git checkout to regenerate drbd/.drbd_git_revision
make[1]: Entering directory `/tmp/pkg/drbd-9.0.23-1/drbd'

    Calling toplevel makefile of kernel source tree, which I believe is in
    KDIR=/lib/modules/3.10.0-1127.10.1.el7.x86_64/build

make -C /lib/modules/3.10.0-1127.10.1.el7.x86_64/build   M=/tmp/pkg/drbd-9.0.23-1/drbd  modules
  COMPAT  before_4_13_kernel_read
  COMPAT  alloc_workqueue_takes_fmt
  COMPAT  blkdev_issue_zeroout_discard
  COMPAT  drbd_release_returns_void
  COMPAT  genl_policy_in_ops
  COMPAT  have_SHASH_DESC_ON_STACK
  COMPAT  have_WB_congested_enum
  COMPAT  have_allow_kernel_signal
  COMPAT  have_atomic_dec_if_positive_linux
  COMPAT  have_atomic_in_flight
  COMPAT  have_bd_claim_by_disk
  COMPAT  have_bd_unlink_disk_holder
  COMPAT  have_bio_bi_bdev
  COMPAT  have_bio_bi_error
  COMPAT  have_bio_bi_opf
  COMPAT  have_bio_bi_status
  COMPAT  have_bio_clone_fast
  COMPAT  have_bio_flush
  COMPAT  have_bio_free
  COMPAT  have_bio_op_shift
  COMPAT  have_bio_set_op_attrs
  COMPAT  have_bio_rw
  COMPAT  have_bioset_create_front_pad
  COMPAT  have_bioset_init
  COMPAT  have_bioset_need_bvecs
  COMPAT  have_blk_check_plugged
  COMPAT  have_blk_qc_t_make_request
  COMPAT  have_blk_queue_flag_set
  COMPAT  have_blk_queue_make_request
  COMPAT  have_blk_queue_merge_bvec
  COMPAT  have_blk_queue_split_q_bio
/bin/bash: /usr/bin/mkdir: Argument list too long
/bin/bash: /usr/bin/tr: Argument list too long
make[3]: execvp: /bin/bash: Argument list too long
make[3]: *** [/tmp/pkg/drbd-9.0.23-1/drbd/.compat_test.3.10.0-1127.10.1.el7.x86_64/have_blk_queue_split_q_bio_bioset.result] Error 127
make[3]: *** Waiting for unfinished jobs....
/bin/bash: /usr/bin/tr: Argument list too long
  COMPAT  have_blk_queue_plugged
/bin/bash: /usr/bin/tr: Argument list too long
/bin/bash: /usr/bin/tr: Argument list too long
make[2]: *** [_module_/tmp/pkg/drbd-9.0.23-1/drbd] Error 2
make[1]: *** [kbuild] Error 2
make[1]: Leaving directory `/tmp/pkg/drbd-9.0.23-1/drbd'
make: *** [module] Error 2

Could not find the expexted *.ko, see stderr for more details
```

I managed to make some progress: I can recreate the bug using:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:50:46Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

```
$ uname -a
Linux centos-7-k8s-200.test 3.10.0-1127.13.1.el7.x86_64 #1 SMP Tue Jun 23 15:46:38 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
  COMPAT  have_bio_start_io_acct                                                                                                                                                     [20/1813]
  COMPAT  have_bioset_create_front_pad
make[3]: execvp: /bin/bash: Argument list too long
make[3]: *** [/tmp/pkg/drbd-9.0.24-1/drbd/.compat_test.3.10.0-1127.13.1.el7.x86_64/have_bioset_init.result] Error 127
make[3]: *** Waiting for unfinished jobs....          
/bin/bash: line 0: echo: write error: Cannot allocate memory
make[3]: *** [/tmp/pkg/drbd-9.0.24-1/drbd/.compat_test.3.10.0-1127.13.1.el7.x86_64/have_bio_free.result] Error 1
make[2]: *** [_module_/tmp/pkg/drbd-9.0.24-1/drbd] Error 2           
make[1]: *** [kbuild] Error 2  
make[1]: Leaving directory `/tmp/pkg/drbd-9.0.24-1/drbd'              
make: *** [module] Error 2                                   
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * `operator.resources` for operator containers
   * `operator.controller.resources` for LINSTOR controller containers
   * `operator.satelliteSet.resources` for LINSTOR satellite containers
+  * `operator.satelliteSet.kernelModuleInjectionResources` for kernel module injector/builder containers
 * Components deployed by the operator can now run with multiple replicas. Components
   elect a leader, that will take on the actual work as long as it is active. Should one
   pod go down, another replica will take over.

--- a/charts/piraeus/crds/piraeus.linbit.com_linstornodesets_crd.yaml
+++ b/charts/piraeus/crds/piraeus.linbit.com_linstornodesets_crd.yaml
@@ -640,6 +640,26 @@ spec:
               description: kernelModImage is the image (location + tag) for the LINSTOR/DRBD
                 kernel module injector container
               type: string
+            kernelModuleInjectionResources:
+              description: Resource requirements for the kernel module builder/injector
+                container
+              nullable: true
+              properties:
+                limits:
+                  additionalProperties:
+                    type: string
+                  description: 'Limits describes the maximum amount of compute resources
+                    allowed. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                  type: object
+                requests:
+                  additionalProperties:
+                    type: string
+                  description: 'Requests describes the minimum amount of compute resources
+                    required. If Requests is omitted for a container, it defaults
+                    to Limits if that is explicitly specified, otherwise to an implementation-defined
+                    value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                  type: object
+              type: object
             linstorHttpsClientSecret:
               description: 'Name of the secret containing: (a) `ca.pem`: root certificate
                 used to validate HTTPS connections with Linstor (PEM format, without
@@ -653,7 +673,7 @@ spec:
                 the node pods
               type: string
             resources:
-              description: Resource requirements for the LINSTOR satellite pod
+              description: Resource requirements for the LINSTOR satellite container
               nullable: true
               properties:
                 limits:

--- a/charts/piraeus/crds/piraeus.linbit.com_linstorsatellitesets_crd.yaml
+++ b/charts/piraeus/crds/piraeus.linbit.com_linstorsatellitesets_crd.yaml
@@ -640,6 +640,26 @@ spec:
               description: kernelModImage is the image (location + tag) for the LINSTOR/DRBD
                 kernel module injector container
               type: string
+            kernelModuleInjectionResources:
+              description: Resource requirements for the kernel module builder/injector
+                container
+              nullable: true
+              properties:
+                limits:
+                  additionalProperties:
+                    type: string
+                  description: 'Limits describes the maximum amount of compute resources
+                    allowed. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                  type: object
+                requests:
+                  additionalProperties:
+                    type: string
+                  description: 'Requests describes the minimum amount of compute resources
+                    required. If Requests is omitted for a container, it defaults
+                    to Limits if that is explicitly specified, otherwise to an implementation-defined
+                    value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                  type: object
+              type: object
             linstorHttpsClientSecret:
               description: 'Name of the secret containing: (a) `ca.pem`: root certificate
                 used to validate HTTPS connections with Linstor (PEM format, without
@@ -653,7 +673,7 @@ spec:
                 the node pods
               type: string
             resources:
-              description: Resource requirements for the LINSTOR satellite pod
+              description: Resource requirements for the LINSTOR satellite container
               nullable: true
               properties:
                 limits:

--- a/charts/piraeus/templates/operator-satelliteset.yaml
+++ b/charts/piraeus/templates/operator-satelliteset.yaml
@@ -16,6 +16,7 @@ spec:
   affinity: {{ .Values.operator.satelliteSet.affinity | toJson }}
   tolerations: {{ .Values.operator.satelliteSet.tolerations | toJson}}
   resources: {{ .Values.operator.satelliteSet.resources | toJson }}
+  kernelModuleInjectionResources: {{ .Values.operator.satelliteSet.kernelModuleInjectionResources | toJson }}
   {{- if .Values.operator.satelliteSet.storagePools }}
   storagePools:
 {{ toYaml .Values.operator.satelliteSet.storagePools | indent 4 }}

--- a/charts/piraeus/values.cn.yaml
+++ b/charts/piraeus/values.cn.yaml
@@ -63,3 +63,4 @@ operator:
     affinity: {}
     tolerations: []
     resources: {}
+    kernelModuleInjectionResources: {}

--- a/charts/piraeus/values.yaml
+++ b/charts/piraeus/values.yaml
@@ -63,3 +63,4 @@ operator:
     affinity: {}
     tolerations: []
     resources: {}
+    kernelModuleInjectionResources: {}

--- a/examples/resource-requirements.yaml
+++ b/examples/resource-requirements.yaml
@@ -39,3 +39,10 @@ operator:
       requests:
         cpu: "0.1"
         memory: "300Mi"
+    kernelModuleInjectionResources:
+      limits:
+        cpu: "1.0"
+        memory: "1Gi"
+      requests:
+        cpu: "0.5"
+        memory: "750Mi"

--- a/pkg/apis/piraeus/v1alpha1/linstorsatelliteset_types.go
+++ b/pkg/apis/piraeus/v1alpha1/linstorsatelliteset_types.go
@@ -67,10 +67,15 @@ type LinstorSatelliteSetSpec struct {
 	// +optional
 	ControllerEndpoint string `json:"controllerEndpoint"`
 
-	// Resource requirements for the LINSTOR satellite pod
+	// Resource requirements for the LINSTOR satellite container
 	// +optional
 	// +nullable
 	Resources corev1.ResourceRequirements `json:"resources"`
+
+	// Resource requirements for the kernel module builder/injector container
+	// +optional
+	// +nullable
+	KernelModuleInjectionResources corev1.ResourceRequirements `json:"kernelModuleInjectionResources"`
 
 	// Affinity for scheduling the satellite pods
 	// +optional

--- a/pkg/apis/piraeus/v1alpha1/zz_generated.deepcopy.go
+++ b/pkg/apis/piraeus/v1alpha1/zz_generated.deepcopy.go
@@ -569,6 +569,7 @@ func (in *LinstorSatelliteSetSpec) DeepCopyInto(out *LinstorSatelliteSetSpec) {
 		**out = **in
 	}
 	in.Resources.DeepCopyInto(&out.Resources)
+	in.KernelModuleInjectionResources.DeepCopyInto(&out.KernelModuleInjectionResources)
 	if in.Affinity != nil {
 		in, out := &in.Affinity, &out.Affinity
 		*out = new(v1.Affinity)

--- a/pkg/controller/linstorsatelliteset/linstorsatelliteset_controller.go
+++ b/pkg/controller/linstorsatelliteset/linstorsatelliteset_controller.go
@@ -896,6 +896,7 @@ func daemonSetWithDRBDKernelModuleInjection(ds *apps.DaemonSet, pns *piraeusv1al
 					MountPath: kubeSpec.ModulesDir,
 				},
 			},
+			Resources: pns.Spec.KernelModuleInjectionResources,
 		},
 	}
 

